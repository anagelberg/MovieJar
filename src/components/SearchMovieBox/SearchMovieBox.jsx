import './SearchMovieBox.scss';
import { useNavigate } from 'react-router-dom';
import { ReactComponent as SearchIcon } from '../../assets/icons/search.svg'



function SearchMovieBox({ modifier }) {

    const navigate = useNavigate();

    const handleSearchBarEntry = (e) => {
        const fieldValue = e.target.value
        if (e.key === 'Enter' && fieldValue.length >= 1) {
            navigate(`/search/${fieldValue}`)
        }
    }

    return (
        <div className={`search search${modifier}`}>
            <SearchIcon className='search__icon' />
            <input
                className={`search__input search__input${modifier}`}
                placeholder="Find movies..."
                onKeyDown={(e) => { handleSearchBarEntry(e) }}
            ></input>
        </div>
    )
}

export default SearchMovieBox;
