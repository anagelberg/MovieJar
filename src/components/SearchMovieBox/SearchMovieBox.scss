@use "../../styles/variables" as *;
@use "../../styles/mixins" as *;

.search {
  width: 100%;
  max-width: $max-form-field-width;
  position: relative;
  margin: auto;
  @include flex-center;

  &--tablet {
    @include tablet-bigger-only;
  }

  &--mobile {
    @include mobile-only;
  }

  &__input {
    @include form-field;
  }

  &__icon {
    position: absolute;
    fill: $dark-text;
    left: 10px; //come back to!
  }
}
