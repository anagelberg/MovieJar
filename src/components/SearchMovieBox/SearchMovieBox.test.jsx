import { screen, render, fireEvent } from '@testing-library/react';
import SearchMovieBox from './SearchMovieBox';
import { MemoryRouter } from 'react-router-dom'; // simulates routing for testing purposes


// Mock the navigate function
const mockNavigate = jest.fn();

jest.mock('react-router-dom', () => ({
    ...jest.requireActual('react-router-dom'), // use actual for all non-hook parts
    useNavigate: () => mockNavigate, // mock useNavigate hook when called
}));



describe("SearchMovieBox component", () => {

    it("renders input element", () => {
        render(
            <MemoryRouter>
                <SearchMovieBox />
            </MemoryRouter>
        )
        expect(screen.getByRole('textbox')).toBeInTheDocument()
    })



    it("renders input element with correct modifier", () => {
        render(
            <MemoryRouter>
                <SearchMovieBox modifier="--large" />
            </MemoryRouter>
        )
        expect(screen.getByRole('textbox')).toHaveClass('search__input--large')
    })


    it("matches snapshot", () => {
        const { asFragment } = render(
            <MemoryRouter>
                <SearchMovieBox />
            </MemoryRouter>
        )
        expect(asFragment()).toMatchSnapshot()
    })


    it("triggers navigation when Enter key is pressed with a non-empty input", () => {
        render(
            <MemoryRouter>
                <SearchMovieBox />
            </MemoryRouter>
        )

        const input = screen.getByRole('textbox')
        fireEvent.change(input, { target: { value: 'test' } })
        fireEvent.keyDown(input, { key: 'Enter', code: 'Enter', keyCode: 13 })

        expect(mockNavigate).toHaveBeenCalledWith('/search/test')
    })


    it("does not trigger navigation when Enter key is pressed with an empty input", () => {
        render(
            <MemoryRouter>
                <SearchMovieBox />
            </MemoryRouter>
        )

        const input = screen.getByRole('textbox')
        fireEvent.keyDown(input, { key: 'Enter', code: 'Enter', keyCode: 13 })

        expect(mockNavigate).not.toHaveBeenCalled()
    })


})

