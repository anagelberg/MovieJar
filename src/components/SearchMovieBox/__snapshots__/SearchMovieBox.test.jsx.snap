// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchMovieBox component matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="search searchundefined"
  >
    <svg
      class="search__icon"
    >
      search.svg
    </svg>
    <input
      class="search__input search__inputundefined"
      placeholder="Find movies..."
    />
  </div>
</DocumentFragment>
`;
